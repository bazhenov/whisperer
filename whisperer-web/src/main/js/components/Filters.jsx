import React, { Component } from 'react'

export default class Filters extends Component {

	constructor(props) {
		super(props);
		this.handleChange = this.handleChange.bind(this);
		this.updateFiltersFilter = this.updateFiltersFilter.bind(this);
		this.removeFilterHandler = this.removeFilterHandler.bind(this);
		this.state = {
			packageNameFilter: '',
			hostNameFilter: '',
			threadNameFilter: ''
		}
	}

	updateFiltersFilter(e) {
		this.setState({
			...this.state,
			[e.target.name]: e.target.value
		})
	}

	removeFilterHandler(name, value, e) {
		this.removeFilter(name, value);
		e.preventDefault();
	}

	removeFilter(name, value) {
		const { filters, updateFilters } = this.props;
		const newFilters = { ...filters, [name]: filters[name].delete(value) };
		updateFilters(newFilters);
	}

	addFilter(name, value) {
		const { filters, updateFilters } = this.props;
		const newFilters = { ...filters, [name]: filters[name].add(value) };
		updateFilters(newFilters);
	}

	handleChange(e) {
		const { name, value, checked } = e.target;
		checked ? this.addFilter(name, value) : this.removeFilter(name, value);
	}

	_getOptions(values, filter) {
		return values.sortBy(
			(k, v) => {return {count: k, value: v}},
			(a, b) => {
				if (a.count === 0 && b.count > 0) return 1;
				if (a.count > 0 && b.count === 0) return -1;
				return a.value.localeCompare(b.value);
			}).filter((k, v) => filter.length === 0 || v.toLowerCase().indexOf(filter.toLowerCase()) > -1);
	}

	_getCheckboxes(options, name) {
		const { filters } = this.props;
		const checkboxes = options.map((count, value) => {
			return <div className="checkbox filters-checkbox" key={value}>
				<label className={count === 0 ? 'zero-count-filter' : ''}>
					<input type="checkbox" name={name} onChange={this.handleChange} value={value}
											checked={filters[name].contains(value)} />{value}
				</label>
				<div className="filters-count">
					<span className="filter-count-label">{count}</span>
				</div>
			</div>
		}).toArray();
		return <div className="filters-filter-options">{checkboxes}</div>
	}

	renderFilterFilter(label, name) {
		return <div className="form-group">
			<label htmlFor={name}>{label}</label>
			<input type="text" className="form-control" id={name} name={name} value={this.state[name]}
						 onChange={this.updateFiltersFilter} />
		</div>
	}

	_filtersIsEmpty() {
		const { packageNames, hostNames, threadNames } = this.props.filtersValues;
		return packageNames.isEmpty() && hostNames.isEmpty() && threadNames.isEmpty();
	}

	_filtersLabels(name) {
		const currentFilter = this.props.filters[name];
		if (currentFilter.isEmpty()) return null;
		const labels = currentFilter.map(v =>
			<div key={v}>
					<span className="filter-label-to-remove label label-info">
						<a type="button" onClick={e => this.removeFilterHandler(name, v, e)} className="close remove-filter">
							<span aria-hidden="true">&times;</span>
						</a>
						{v}
					</span>
			</div>
		);
		return <div className="filter-labels">{ labels }</div>;
	}

	renderFilter(label, filterName, field) {
		const options = this._getOptions(this.props.filtersValues[field], this.state[filterName]);
		return <div className="col-xs-4">
			{this.renderFilterFilter(label, filterName)}
			{this._getCheckboxes(options, field)}
			{this._filtersLabels(field)}
		</div>
	}

	render() {
		if (this._filtersIsEmpty()) return null;
		return <div className="well" id="filters">
			<div className="row">
				{this.renderFilter('Package Name', 'packageNameFilter', 'packageNames')}
				{this.renderFilter('Host Name', 'hostNameFilter', 'hostNames')}
				{this.renderFilter('Thread Name', 'threadNameFilter', 'threadNames')}
			</div>
		</div>
	}
}