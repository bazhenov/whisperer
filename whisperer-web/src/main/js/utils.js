import { Set, List, Map } from 'immutable';

export const getGroupShortName = (group) => {
	const parts = (group || '').split('.');
	return parts.reduce((name, part, i) => {
		if (part.length === 0) return name;
		return name + (name.length === 0 ? '' : '.') + (i === (parts.length - 1) ? part : part[0])
	}, '');
};

const isEmptyOrContains = (col, value) => {
	return col.isEmpty() || col.contains(value);
};

export const isMessageSuitedForFilters = (m, filters) => {
	return isEmptyOrContains(filters.packageNames, m.shortGroupName) &&
		isEmptyOrContains(filters.hostNames, m.host) &&
		isEmptyOrContains(filters.threadNames, m.thread);
};

export const filterMessages = (messages, filters) => {
	return messages.filter(m => isMessageSuitedForFilters(m, filters));
};

export const emptyFilters = () => {
	return buildFilters(Set(), Set(), Set());
};

export const buildFilters = (hostNames, packageNames, threadNames) => {
	return { hostNames, packageNames, threadNames };
};

export const emptyConnectionParams = () => {
	return {
		key: '', value: '', prefix: '', severity: 'INFO'
	};
};

export const newConnectionParams = (key, value, prefix, severity) => {
	return { key, value, prefix, severity };
};

export const updateFiltersValues = (current, filters, message) => {

	const updateFilterValues = (name, value) => {
		let res = current[name];
		if (isMessageSuitedForFilters(message, { ...filters, [name]: new Set() })) {
			res = res.update(value, 0, v => v + 1);
		}
		res = res.has(value) ? res : res.set(value, 0);
		return res;
	};
	return {
		hostNames: updateFilterValues('hostNames', message.host),
		packageNames: updateFilterValues('packageNames', message.shortGroupName),
		threadNames: updateFilterValues('threadNames', message.thread)
	};
};

export const createFiltersValues = (all, filters) => {
	return all.reduce((filtersValues, m) => updateFiltersValues(filtersValues, filters, m), emptyFilterValues());
};

export const emptyFilterValues = () => {
	return buildFilterValues(Map(), Map(), Map());
};

export const buildFilterValues = (hostNames, packageNames, threadNames) => {
	return { hostNames, packageNames, threadNames };
};

export const emptyMessages = () => {
	return {
		all: new List(),
		filtered: new List(),
		filters: emptyFilters(),
		filtersValues: emptyFilterValues(),
		nextMessageId: 1
	}
};