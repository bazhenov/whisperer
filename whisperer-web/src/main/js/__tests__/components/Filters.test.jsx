import React from "react";
import Filters from "../../components/Filters.jsx";
import renderer from "react-test-renderer";
import { buildFilters, buildFilterValues, emptyFilterValues } from '../../utils';
import { Set, Map } from 'immutable'
import {mount} from "enzyme";
import toJson from 'enzyme-to-json';

describe('Filters', () => {
	const props = {
		filters: buildFilters(
			Set(['search-coord2.akod.loc', 'search-coord3.akod.loc']),
			Set(['c.f.s.SearchEngine', 'c.f.s.EngineContext']),
			Set(['qtp-1', 'qtp-2', 'qtp-3', 'qtp-4'])
		),
		filtersValues: buildFilterValues(
			Map({
				'search-coord2.akod.loc': 5,
				'search-coord3.akod.loc': 3,
				'search-coord4.akod.loc': 1,
				'search-coord5.akod.loc': 0
			}),
			Map({
				'c.f.s.SearchEngine': 5,
				'c.f.s.EngineContext': 3,
				'a.n.o.ther': 0
			}),
			Map({
				'qtp-1': 5,
				'qtp-2': 3,
				'qtp-3': 0
			}),
		),
		updateFilters: () => {}
	};
	it('rendering', () => {
		expect(renderer.create(
			<Filters { ...props } />
		).toJSON()).toMatchSnapshot();

		expect(renderer.create(
			<Filters { ...props } filtersValues={emptyFilterValues()} />
		).toJSON()).toMatchSnapshot();
	});

	describe('interactions', () => {
		it('adding filter', done => {
			const updateFilters = filters => {
				expect(filters).toEqual(
					buildFilters(
						props.filters.hostNames.add('search-coord4.akod.loc'),
						props.filters.packageNames,
						props.filters.threadNames,
					)
				);
				done();
			};
			mount(
				<Filters { ...props } updateFilters={updateFilters} />
			).find('input')
				.filterWhere(chk => chk.props().value === 'search-coord4.akod.loc')
				.simulate('change',{ target: { checked: true, name: 'hostNames', value: 'search-coord4.akod.loc' } });
		});
		it('removing filter', done => {
			const updateFilters = filters => {
				expect(filters).toEqual(
					buildFilters(
						props.filters.hostNames.remove('search-coord3.akod.loc'),
						props.filters.packageNames,
						props.filters.threadNames,
					)
				);
				done();
			};
			mount(
				<Filters { ...props } updateFilters={updateFilters} />
			).find('input')
				.filterWhere(chk => chk.props().value === 'search-coord3.akod.loc')
				.simulate('change',{ target: { checked: false, name: 'hostNames', value: 'search-coord3.akod.loc' } });
		});
		it('filters filtering', () => {
			const wrapper = mount(
				<Filters { ...props } />
			);
			wrapper.find('input')
				.filterWhere(chk => chk.props().name === 'packageNameFilter')
				.simulate('change', { target: { name: 'packageNameFilter', value: 'c.f.s' } });
			expect(toJson(wrapper)).toMatchSnapshot();
		});
	});
});