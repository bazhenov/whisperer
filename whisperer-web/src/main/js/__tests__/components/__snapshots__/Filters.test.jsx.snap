// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters interactions filters filtering 1`] = `
<Filters
  filters={
    Object {
      "hostNames": Array [
        "search-coord2.akod.loc",
        "search-coord3.akod.loc",
      ],
      "packageNames": Array [
        "c.f.s.SearchEngine",
        "c.f.s.EngineContext",
      ],
      "threadNames": Array [
        "qtp-1",
        "qtp-2",
        "qtp-3",
        "qtp-4",
      ],
    }
  }
  filtersValues={
    Object {
      "hostNames": Object {
        "search-coord2.akod.loc": 5,
        "search-coord3.akod.loc": 3,
        "search-coord4.akod.loc": 1,
        "search-coord5.akod.loc": 0,
      },
      "packageNames": Object {
        "a.n.o.ther": 0,
        "c.f.s.EngineContext": 3,
        "c.f.s.SearchEngine": 5,
      },
      "threadNames": Object {
        "qtp-1": 5,
        "qtp-2": 3,
        "qtp-3": 0,
      },
    }
  }
  updateFilters={[Function]}
>
  <div>
    <h3>
      Filters
    </h3>
    <div
      className="row"
    >
      <div
        className="col-xs-4"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="packageNameFilter"
          >
            Package Name
          </label>
          <input
            className="form-control"
            id="packageNameFilter"
            name="packageNameFilter"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="filters-filter-options"
        >
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className=""
            >
              <input
                checked={true}
                name="packageNames"
                onChange={[Function]}
                type="checkbox"
                value="c.f.s.EngineContext"
              />
              c.f.s.EngineContext
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                3
              </span>
            </div>
          </div>
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className=""
            >
              <input
                checked={true}
                name="packageNames"
                onChange={[Function]}
                type="checkbox"
                value="c.f.s.SearchEngine"
              />
              c.f.s.SearchEngine
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                5
              </span>
            </div>
          </div>
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className="zero-count-filter"
            >
              <input
                checked={false}
                name="packageNames"
                onChange={[Function]}
                type="checkbox"
                value="a.n.o.ther"
              />
              a.n.o.ther
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                0
              </span>
            </div>
          </div>
        </div>
        <div
          className="filter-labels"
        >
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              c.f.s.SearchEngine
            </span>
          </div>
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              c.f.s.EngineContext
            </span>
          </div>
        </div>
      </div>
      <div
        className="col-xs-4"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="hostNameFilter"
          >
            Host Name
          </label>
          <input
            className="form-control"
            id="hostNameFilter"
            name="hostNameFilter"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="filters-filter-options"
        >
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className=""
            >
              <input
                checked={true}
                name="hostNames"
                onChange={[Function]}
                type="checkbox"
                value="search-coord2.akod.loc"
              />
              search-coord2.akod.loc
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                5
              </span>
            </div>
          </div>
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className=""
            >
              <input
                checked={true}
                name="hostNames"
                onChange={[Function]}
                type="checkbox"
                value="search-coord3.akod.loc"
              />
              search-coord3.akod.loc
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                3
              </span>
            </div>
          </div>
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className=""
            >
              <input
                checked={false}
                name="hostNames"
                onChange={[Function]}
                type="checkbox"
                value="search-coord4.akod.loc"
              />
              search-coord4.akod.loc
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                1
              </span>
            </div>
          </div>
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className="zero-count-filter"
            >
              <input
                checked={false}
                name="hostNames"
                onChange={[Function]}
                type="checkbox"
                value="search-coord5.akod.loc"
              />
              search-coord5.akod.loc
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                0
              </span>
            </div>
          </div>
        </div>
        <div
          className="filter-labels"
        >
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              search-coord2.akod.loc
            </span>
          </div>
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              search-coord3.akod.loc
            </span>
          </div>
        </div>
      </div>
      <div
        className="col-xs-4"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="threadNameFilter"
          >
            Thread Name
          </label>
          <input
            className="form-control"
            id="threadNameFilter"
            name="threadNameFilter"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="filters-filter-options"
        >
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className=""
            >
              <input
                checked={true}
                name="threadNames"
                onChange={[Function]}
                type="checkbox"
                value="qtp-1"
              />
              qtp-1
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                5
              </span>
            </div>
          </div>
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className=""
            >
              <input
                checked={true}
                name="threadNames"
                onChange={[Function]}
                type="checkbox"
                value="qtp-2"
              />
              qtp-2
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                3
              </span>
            </div>
          </div>
          <div
            className="checkbox filters-checkbox"
          >
            <label
              className="zero-count-filter"
            >
              <input
                checked={true}
                name="threadNames"
                onChange={[Function]}
                type="checkbox"
                value="qtp-3"
              />
              qtp-3
            </label>
            <div
              className="filters-count"
            >
              <span
                className="label label-default"
              >
                0
              </span>
            </div>
          </div>
        </div>
        <div
          className="filter-labels"
        >
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              qtp-1
            </span>
          </div>
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              qtp-2
            </span>
          </div>
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              qtp-3
            </span>
          </div>
          <div>
            <span
              className="filter-label-to-remove label label-info"
            >
              <a
                className="close remove-filter"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </a>
              qtp-4
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</Filters>
`;

exports[`Filters rendering 1`] = `
<div>
  <h3>
    Filters
  </h3>
  <div
    className="row"
  >
    <div
      className="col-xs-4"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="packageNameFilter"
        >
          Package Name
        </label>
        <input
          className="form-control"
          id="packageNameFilter"
          name="packageNameFilter"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="filters-filter-options"
      >
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className=""
          >
            <input
              checked={true}
              name="packageNames"
              onChange={[Function]}
              type="checkbox"
              value="c.f.s.EngineContext"
            />
            c.f.s.EngineContext
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              3
            </span>
          </div>
        </div>
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className=""
          >
            <input
              checked={true}
              name="packageNames"
              onChange={[Function]}
              type="checkbox"
              value="c.f.s.SearchEngine"
            />
            c.f.s.SearchEngine
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              5
            </span>
          </div>
        </div>
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className="zero-count-filter"
          >
            <input
              checked={false}
              name="packageNames"
              onChange={[Function]}
              type="checkbox"
              value="a.n.o.ther"
            />
            a.n.o.ther
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              0
            </span>
          </div>
        </div>
      </div>
      <div
        className="filter-labels"
      >
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            c.f.s.SearchEngine
          </span>
        </div>
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            c.f.s.EngineContext
          </span>
        </div>
      </div>
    </div>
    <div
      className="col-xs-4"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="hostNameFilter"
        >
          Host Name
        </label>
        <input
          className="form-control"
          id="hostNameFilter"
          name="hostNameFilter"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="filters-filter-options"
      >
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className=""
          >
            <input
              checked={true}
              name="hostNames"
              onChange={[Function]}
              type="checkbox"
              value="search-coord2.akod.loc"
            />
            search-coord2.akod.loc
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              5
            </span>
          </div>
        </div>
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className=""
          >
            <input
              checked={true}
              name="hostNames"
              onChange={[Function]}
              type="checkbox"
              value="search-coord3.akod.loc"
            />
            search-coord3.akod.loc
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              3
            </span>
          </div>
        </div>
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className=""
          >
            <input
              checked={false}
              name="hostNames"
              onChange={[Function]}
              type="checkbox"
              value="search-coord4.akod.loc"
            />
            search-coord4.akod.loc
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              1
            </span>
          </div>
        </div>
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className="zero-count-filter"
          >
            <input
              checked={false}
              name="hostNames"
              onChange={[Function]}
              type="checkbox"
              value="search-coord5.akod.loc"
            />
            search-coord5.akod.loc
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              0
            </span>
          </div>
        </div>
      </div>
      <div
        className="filter-labels"
      >
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            search-coord2.akod.loc
          </span>
        </div>
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            search-coord3.akod.loc
          </span>
        </div>
      </div>
    </div>
    <div
      className="col-xs-4"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="threadNameFilter"
        >
          Thread Name
        </label>
        <input
          className="form-control"
          id="threadNameFilter"
          name="threadNameFilter"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="filters-filter-options"
      >
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className=""
          >
            <input
              checked={true}
              name="threadNames"
              onChange={[Function]}
              type="checkbox"
              value="qtp-1"
            />
            qtp-1
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              5
            </span>
          </div>
        </div>
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className=""
          >
            <input
              checked={true}
              name="threadNames"
              onChange={[Function]}
              type="checkbox"
              value="qtp-2"
            />
            qtp-2
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              3
            </span>
          </div>
        </div>
        <div
          className="checkbox filters-checkbox"
        >
          <label
            className="zero-count-filter"
          >
            <input
              checked={true}
              name="threadNames"
              onChange={[Function]}
              type="checkbox"
              value="qtp-3"
            />
            qtp-3
          </label>
          <div
            className="filters-count"
          >
            <span
              className="label label-default"
            >
              0
            </span>
          </div>
        </div>
      </div>
      <div
        className="filter-labels"
      >
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            qtp-1
          </span>
        </div>
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            qtp-2
          </span>
        </div>
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            qtp-3
          </span>
        </div>
        <div>
          <span
            className="filter-label-to-remove label label-info"
          >
            <a
              className="close remove-filter"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </a>
            qtp-4
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters rendering 2`] = `null`;
